src/zen/workspaces/ZenWorkspaces.mjs

```diff
@@ -2275,6 +2275,14 @@ var gZenWorkspaces = new (class extends ZenMultiWindowFeature {
     if (!activeWorkspace) {
       return;
     }
+
+    // This check may be code smell, but without it, the tab will open in the properly
+    // routed workspace, and then the workspace gets switched back to the previously
+    // active workspace.
+    if (tab.hasAttribute('zen-atc-created')) {
```

Comment:
Should this attribute get removed once checked? Because once the tab is created, it wont pass this stage anymore

Reply:
Done!